Tools/autotest/arducopter.py

```diff
@@ -4416,6 +4416,48 @@ def PayLoadPlaceMission(self):
         if ex is not None:
             raise ex
 
+    def weathervane_test(self):
+        # We test nose into wind code paths and yaw direction here and test side into wind
+        # yaw direction in QuadPlane tests to reduce repetition.
+        self.set_parameters({"SIM_WIND_SPD": 10,
+                             "SIM_WIND_DIR": 100,
+                             "GUID_OPTIONS": 129, # allow weathervaning and arming from tx in guided
+                             "AUTO_OPTIONS": 131, # allow arming in auto, take off without raising the stick, and weathervaning
+                             "WVANE_ENABLE": 1,
+                             "WVANE_GAIN": 3,
+                             "WVANE_VELZ_MAX": 1,
+                             "WVANE_SPD_MAX": 2})
+
+        self.progress("Test weathervaning in auto")
+        num_wp = self.load_mission("weathervane_mission.txt", strict=False)
+        if not num_wp:
+            raise NotAchievedException("load weathervane_mission failed")
```

Comment:
```suggestion
```

not required.... if this check is really needed it should be in `load_mission`

Reply:
done